sessionspaces/src/permissionables/session.rs

```diff
@@ -0,0 +1,77 @@
+use sqlx::{query_as, MySqlPool};
+use tracing::instrument;
+
+/// A singular beamline session
+pub struct Session {
+    /// The opaque identifier of the session
+    pub id: u32,
+    /// The proposal type code
+    pub code: String,
+    /// The proposal number with which this session is associated
+    pub proposal: u32,
+    /// The visit number within the proposal
+    pub visit: u32,
+}
+
+impl Session {
+    /// Fetches [`Session`]s from ISPyB
+    #[instrument(name = "fetch_sessions")]
+    pub async fn fetch(ispyb_pool: &MySqlPool) -> Result<Vec<Self>, sqlx::Error> {
```

Comment:
This needs to be tested. Can do it with a mock database instance as I did [in the authz bundler](https://github.com/DiamondLightSource/authz/blob/6539c4d95bb83724bfdf4ea1b54f7040efe5e191/bundler/src/permissionables/sessions.rs#L103)